addons/account_facturx/models/ir_actions_report.py

```diff
@@ -17,16 +19,31 @@ def _post_pdf(self, save_in_attachment, pdf_content=None, res_ids=None):
             if invoice.is_sale_document() and invoice.state != 'draft':
                 xml_content = invoice._export_as_facturx_xml()
 
-                # Add attachment.
-                reader_buffer = io.BytesIO(pdf_content)
+                reader_buffer = BytesIO(pdf_content)
                 reader = PdfFileReader(reader_buffer)
-                writer = PdfFileWriter()
+                writer = OdooPdfFileWriter()
                 writer.cloneReaderDocumentRoot(reader)
-                writer.addAttachment('factur-x.xml', xml_content)
-                buffer = io.BytesIO()
+
+                if tools.str2bool(self.env['ir.config_parameter'].sudo().get_param('edi.use_pdfa', 'False')):
+                    try:
+                        writer.convert_to_pdfa()
+                    except Exception as e:
+                        _logger.exception("Error while converting to PDF/A: %s" % str(e))
```

Comment:
Not very important, but it is a best practice to avoid doing string formatting yourself when emitting logs, and instead leave that to the logger[1]: you simply pass the arguments after the log message. That makes the code a bit prettier, and among other things, it means the formatting cost does not need to be paid if the log message is silenced.

Besides, `%s` format always performs `str()` on its argument, so you don't need to do it at all.

```suggestion
                        _logger.exception("Error while converting to PDF/A: %s", e)
```

(The exception message will be printed twice because `logger.exception()` does it too, but that's fine, at least here it will be on the same line)

[1] https://docs.python.org/3/library/logging.html#logging.debug

Reply:
Done thank you :)